### DISCUSSION TITLE
Units difference between observations and forecast

### DISCUSSION THREAD

**Discussion by txhammer68** [weather-gov/api#643] [General]
So i am using this link for current observations 
https://api.weather.gov/stations/KHOU/observations/latest

Works fine but units are in metric, tried adding ?units=us to url, but then get bad request error

Using this link for forecast data
https://api.weather.gov/gridpoints/HGX/76,92/forecast
Works great, results are reported in imperial, US standards.

Any fix for observations to get units in imperial standards?

Also what time frame is used for day/night forecasts? 7am - 7pm ?
Is the refresh duration for current observations every hour?, this seems like too long, used to every 10-15 mins

**Comment by dwhitemv25** [comment]
`/observations` is using QuantitativeValue, so units for the measurements are provided. 

`/gridpoints/.../forecast` defaults to undecorated imperial values, but you can set Feature-Flags to have the temperature and wind values sent as QuantitativeValue instead. 

QV values are always metric; it's assumed you will do your own conversion to imperial, if needed. As a bonus, QV values have increased precision, usually 1 digit after the decimal.

Day/night forecasts break at 6am/6pm local time. You can check the `startTime` and `endTime` properties for each period to confirm.

NDFD data points are only at 1-hour granularity at best (some elements, like QPF, are longer). If they are changing more frequently then that then your WFO actively making changes to the NDFD data while you are watching. This could happen if there is active weather in your area. Otherwise the data is updated on 6-12 hour cycles, depending on when your WFO uploads the forecast packages. There is also CDN caching effects since the forecast documents are set to a 1 hour expiration at the HTTP level.
